src/app/libp2p_helper/src/codanet.go

```diff
@@ -516,7 +516,18 @@ func (h *Helper) handlePxStreams(s network.Stream) {
 
 func (h *Helper) handleTelemetryStreams(s network.Stream) {
 	defer func() {
-		_ = s.Close()
+		err := s.Close()
+		if err != nil {
+			logger.Error("failed to close write side of stream", err)
+			return
+		}
+
+		<-time.After(400 * time.Millisecond)
+
+		err = s.Reset()
```

Comment:
what does reset do?

Reply:
From my understanding of the documentation, reset tells the other side of the stream to hangup and go away. Closing a stream closes our side of the stream, but the connection doesn't get GC'd until the other side closes as well. A reset forcibly tells the other side to close and directly drops the connection.